.github/actions/setup-python/action.yml

```diff
@@ -0,0 +1,43 @@
+name: Setup Python and install requirements
+inputs:
+  python-version:
+    required: true
+  architecture:
+    default: x64
+runs:
+  using: composite
+  steps:
+    # TODO: GH#44980 https://github.com/pypa/setuptools/issues/2941
+    - name: Create temporary requirements.txt
+      run: |
+        # Drop cache at least once per month
+        month_today="$(date '+%Y-%m')"
+        cat > requirements.txt ci/deps/pip-requirements.txt - <<< "# $month_today"
+      shell: bash -el {0}
+
+    - name: Set up Python
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ inputs.python-version }}
+        architecture: ${{ inputs.architecture }}
+        cache: pip
+
+    - name: Fix $PATH on macOS
+      run: |
+        # On macOS, the Python version we installed above is too late in $PATH
+        # to be effective if using "bash -l" (which we need for code that works
+        # with Conda envs).
+        cat >> ~/.bash_profile <<EOF
+          export PATH="\$(echo "\$PATH" | grep -Eio '[^:]+hostedtoolcache/python[^:]+bin'):\$PATH" \
+        EOF
+      shell: bash -el {0}
+      if: ${{ runner.os == 'macOS' }}
+
+    - name: Install dependencies
+      run: |
+        cat requirements.txt
+        # TODO https://github.com/numpy/numpy/issues/21196
+        bits32=$(python -c 'import sys; print(int(sys.maxsize > 2**32))')
```

Comment:
umm what is this?

Reply:
It is required for 32 bit builds but we can add that back later